### PR TITLE
Fixes #15: Add Azure build and push to ACR workflow

### LINKED ISSUE #15: Issue: Add Azure build and push to ACR workflow

### PR DESCRIPTION
**Pull request by redat97**
#15

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,6 +1,5 @@
-# Dockerfile
 # Base image with Python 3.11 on Alpine
-FROM python:3.11-alpine
+FROM python:3.11
```
Comment (rngadam) [inline Dockerfile:2]: why this change?

File: Dockerfile [hunk 2]
```diff
@@ -10,7 +9,8 @@ COPY ./requirements-production.txt requirements.txt
 RUN pip3 install --no-cache-dir -r requirements.txt
 
 # Copy source code into the working directory
-COPY ./src .
+COPY ./app ./app
+COPY finesse-weights.json .
 
 # Use Gunicorn as the server, configuring it for the Flask app
-ENTRYPOINT gunicorn --bind :$PORT --workers 1 --threads 8 --timeout 0 --forwarded-allow-ips "*" app:app
+ENTRYPOINT gunicorn --bind :$PORT --workers 1 --threads 8 --timeout 0 --forwarded-allow-ips="*" app:app
```
Comment (rngadam) [inline Dockerfile:16]: why this change?

File: Dockerfile (hunk 1 above)
Comment (rngadam) [inline Dockerfile:2]: why this change?

File: docs/DEPLOYMENT.md [hunk 3]
```diff
@@ -1,7 +1,8 @@
 # Deployment
 
 ## Azure
-Create and set the environment variables based on `.env.template`. Make sure `FINESSE_BACKEND_DEBUG_MODE` is not set to `True`.
+Create and set the environment variables based on `.env.template`. Make sure 
+`FINESSE_BACKEND_DEBUG_MODE` is not set to `True`.
```
Comment (rngadam) [inline docs/DEPLOYMENT.md:5]: why?

**Review by rngadam — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (redat97) [reply Dockerfile:2]: this was fixed by @k-allagbe I just squashed his commit

File: docs/DEPLOYMENT.md (hunk 3 above)
Comment (redat97) [reply docs/DEPLOYMENT.md:5]: @k-allagbe as well

File: Dockerfile (hunk 1 above)
Comment (k-allagbe) [reply Dockerfile:2]: `ailab-db`, which this app depends on, requires tools that are not present in the `alpine` variant for it's dependencies installation. The tools could be installed on top of `alpine` via the dockerfile but they take very long to build, which could be painful in github actions. I bumped the variant all the way up to avoid all that.

File: docs/DEPLOYMENT.md (hunk 3 above)
Comment (k-allagbe) [reply docs/DEPLOYMENT.md:5]: Why the change?: to keep 80 char max lines.
Why that directive: So that we don't deploy in debug mode.

File: Dockerfile (hunk 2 above)
Comment (redat97) [reply Dockerfile]: my mistake changed

File: docs/DEPLOYMENT.md (hunk 3 above)
Comment (rngadam) [reply docs/DEPLOYMENT.md:5]: it may be that we want to deploy a particular dev wrong in debug mode; I would explain that this should not be set or set to False for production version.

File: Dockerfile (hunk 1 above)
Comment (k-allagbe) [reply Dockerfile:3]: Done.

File: docs/DEPLOYMENT.md (hunk 3 above)
Comment (k-allagbe) [reply docs/DEPLOYMENT.md]: Done.

**Review by k-allagbe — CHANGES_REQUESTED** [review]
Failed deployment on last commit https://github.com/ai-cfia/finesse-backend/pull/17/commits/f1f1c1663e577fd8484b96ce90c49227990901a9

**Comment by redat97** [comment]
> Failed deployment on last commit [f1f1c16](https://github.com/ai-cfia/finesse-backend/commit/f1f1c1663e577fd8484b96ce90c49227990901a9)

fixed

**Review by k-allagbe — APPROVED** [review]

**Review by rngadam — APPROVED** [review]
